### PR TITLE
Update ELI5 What is IBC link

### PR DESCRIPTION
**Pull request by murataniloener**
Documentation content update for the landing page.

This PR fixes a link on the landing page.

### Change scope

The changes in this Pull Request include (please tick all that apply):

* [ ] Small language/grammar fixes
* [X] Small content fixes 
* [ ] Addition of new content
* [ ] Sample code/command updates
* [ ] Platform fixes
* [ ] Other

### CODE REVIEW

**Review by CitMC — APPROVED** [review]
Ready to be merged

**Review by coldice — APPROVED** [review]
